diff --git a/fsleyes/__init__.py b/fsleyes/__init__.py
index 585f60c0..2ba2eda1 100644
--- a/fsleyes/__init__.py
+++ b/fsleyes/__init__.py
@@ -214,7 +214,7 @@ log = None
 
 
 # If set to True, logging will not be configured
-disableLogging = False
+disableLogging = True
 
 
 __version__ = version.__version__
diff --git a/fsleyes/gl/__init__.py b/fsleyes/gl/__init__.py
index f98fe084..34b38cfb 100644
--- a/fsleyes/gl/__init__.py
+++ b/fsleyes/gl/__init__.py
@@ -208,8 +208,8 @@ OpenGL.ERROR_ON_COPY = True
 
 # These flags should be set to True
 # for development, False for production
-OpenGL.ERROR_CHECKING = True
-OpenGL.ERROR_LOGGING  = True
+OpenGL.ERROR_CHECKING = False
+OpenGL.ERROR_LOGGING  = False
 
 
 # If FULL_LOGGING is enabled,
