diff --git a/fsleyes/gl/__init__.py b/fsleyes/gl/__init__.py
index f98fe084..adc06e93 100644
--- a/fsleyes/gl/__init__.py
+++ b/fsleyes/gl/__init__.py
@@ -187,6 +187,7 @@ been created.
 import os
 import sys
 import logging
+import ctypes.util as ctypes_util
 import platform

 import fsl.utils.idle                     as idle
@@ -195,6 +196,22 @@ from   fsl.utils.platform import platform as fslplatform
 import fsleyes_widgets                    as fwidgets


+real_find_library = ctypes_util.find_library
+
+
+def find_library(name):
+    if name in ('OpenGL', 'GLUT'):
+        return f'/System/Library/Frameworks/{name}.framework/{name}'
+    else:
+        return real_find_library(name)
+
+
+# Big sur support was added in Python 3.8.10 / 3.9.1
+pyver = sys.version_info[:3]
+if platform.system() == 'Darwin' and ((pyver < (3, 8, 10)) or pyver == (3, 9, 0)):
+    ctypes_util.find_library = find_library
+
+
 log = logging.getLogger(__name__)
