diff --git a/fsleyes/gl/__init__.py b/fsleyes/gl/__init__.py
index e55fc7ce..1cae93a2 100644
--- a/fsleyes/gl/__init__.py
+++ b/fsleyes/gl/__init__.py
@@ -188,6 +188,7 @@ been created.
 import os
 import sys
 import logging
+import ctypes.util as ctypes_util
 import platform
 
 from   fsl.utils                      import idle
@@ -199,6 +200,22 @@ from   fsleyes.utils                  import lazyimport
 import OpenGL  # noqa
 
 
+real_find_library = ctypes_util.find_library
+
+
+def find_library(name):
+    if name in ('OpenGL', 'GLUT'):
+        return f'/System/Library/Frameworks/{name}.framework/{name}'
+    else:
+        return real_find_library(name)
+
+
+# Big sur support was added in Python 3.8.10 / 3.9.1
+pyver = sys.version_info[:3]
+if platform.system() == 'Darwin' and ((pyver < (3, 8, 10)) or pyver == (3, 9, 0)):
+    ctypes_util.find_library = find_library
+
+
 log = logging.getLogger(__name__)
 
 
