diff --git a/fsleyes/gl/__init__.py b/fsleyes/gl/__init__.py
index 4fb63b5d..2215dbdd 100644
--- a/fsleyes/gl/__init__.py
+++ b/fsleyes/gl/__init__.py
@@ -187,6 +187,7 @@ been created.
 import os
 import sys
 import logging
+import ctypes.util as ctypes_util
 import platform
 
 import fsl.utils.idle                     as idle
@@ -195,6 +196,19 @@ from   fsl.utils.platform import platform as fslplatform
 import fsleyes_widgets                    as fwidgets
 
 
+real_find_library = ctypes_util.find_library
+
+
+def find_library(name):
+    if platform.system() == 'Darwin' and name in ('OpenGL', 'GLUT'):
+        return f'/System/Library/Frameworks/{name}.framework/{name}'
+    else:
+        return real_find_library(name)
+
+
+ctypes_util.find_library = find_library
+
+
 log = logging.getLogger(__name__)
 
 
@@ -246,18 +260,31 @@ def _selectPyOpenGLPlatform():
     # If no display, osmesa on all platforms
     if not fwidgets.canHaveGui():
         os.environ['PYOPENGL_PLATFORM'] = 'osmesa'
+        return
 
-    # GTK3 versions of wxpython 4.1.1 and newer
-    # default to using EGL for GL initialisation,
-    # but pyopengl defaults to glx.
-    elif fslplatform.os.lower() == 'linux':
-        wxver  = fwidgets.wxVersion()
-        wxplat = fwidgets.wxPlatform()
+    # We only need special handling on linux
+    if not fslplatform.os.lower() == 'linux':
+        return
+
+    wxver  = fwidgets.wxVersion()
+    wxplat = fwidgets.wxPlatform()
+
+    # Don't know what version of wxpython
+    # we have - don't do anything
+    if wxver is None:
+        return
+    # GTK3 versions of wxpython>=4.1.1 use EGL
+    # for GL initialisation. All older wxpython
+    # versions, and all GTK2 versions, use GLX.
+    #
+    # PyOpenGL>=3.1.6 will use EGL if it detects
+    # that it is running under Wayland.  But this
+    # breaks things if we have a GTK2 wxpython.
+    if wxplat == fwidgets.GTK2:
+        os.environ['PYOPENGL_PLATFORM'] = 'x11'
+    elif fslversion.compareVersions(wxver, '4.1.1') >= 0:
+        os.environ['PYOPENGL_PLATFORM'] = 'egl'
 
-        if wxver is not None                               and \
-           fslversion.compareVersions(wxver, '4.1.1') >= 0 and \
-           wxplat == fwidgets.WX_GTK3:
-            os.environ['PYOPENGL_PLATFORM'] = 'egl'
 
 
 _selectPyOpenGLPlatform()
